cmd/acr/purge.go

```diff
@@ -211,6 +200,56 @@ func purgeTags(ctx context.Context, acrClient api.AcrCLIClientInterface, poolSiz
 	return deletedTagsCount, nil
 }
 
+// collectTagFilters collects all matching repos and collects the associated tag filters
+func collectTagFilters(ctx context.Context, rawFilters []string, client acrapi.BaseClientAPI) (map[string]string, error) {
+	allRepoNames, err := client.GetRepositories(ctx, "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	tagFilters := map[string]string{}
+	for _, filter := range rawFilters {
+		repoRegex, tagRegex, err := getRepositoryAndTagRegex(filter)
+		if err != nil {
+			return nil, err
+		}
+		var repoNames []string
+		if repoRegex == "" {
+			repoNames = *allRepoNames.Names
```

Comment:
This looks dangerous... People may purge all the repositories/tags by accidentally passing `":"`, `"repo:"`, or `":tags"`. Should we not allow empty repoRegex and empty tagRegex? Is it originally intended? /cc @gildardogmsft

Reply:
This variant was related to the following comment: https://github.com/Azure/acr-cli/issues/57#issuecomment-777899936
I found this way for specifying all repos pretty convenient :)

Reply:
Hmmm I agree with @Wwwsylvia having an empty regex meaning all repositories sounds a little bit dangerous to me, I would prefer to not allow empty repoRegex and also not allow empty tagRegex

Reply:
Added errors on missing tag or repo regex

Reply:
![image](https://user-images.githubusercontent.com/2405792/153743855-241f0c25-0aae-4680-9da2-6294e2379fd5.png)